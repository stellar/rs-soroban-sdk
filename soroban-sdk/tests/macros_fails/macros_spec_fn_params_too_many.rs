use soroban_sdk::contractimpl;

pub struct Contract;

#[contractimpl]
impl Contract {
    pub fn pay(
        p1: u32,
        p2: u32,
        p3: u32,
        p4: u32,
        p5: u32,
        p6: u32,
        p7: u32,
        p8: u32,
        p9: u32,
        p10: u32,
        p11: u32,
    ) {
        unimplemented!()
    }
}

pub fn main() {}
