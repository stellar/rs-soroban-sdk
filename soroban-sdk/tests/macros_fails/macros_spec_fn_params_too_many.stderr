error: contract function has too many parameters, max count 10 parameters
  --> tests/macros_fails/macros_spec_fn_params_too_many.rs:18:9
   |
18 |         p11: u32,
   |         ^^^
